docs/deployment-guide.md

```diff
@@ -0,0 +1,29 @@
+# Deploying Container Object Storage Interface (COSI) On Kubernetes
+
+This document describes steps for Kubernetes administrators to setup Container Object Storage Interface (COSI) onto a Kubernetes cluster.
+
+## Overview
+
+There are four components that need to be deployed in Kubernetes to setup COSI.
+
+- CustomResourceDefinitions (CRDs)
+- Controller
+- Provisioner Sidecar
```

Comment:
- Driver 
- Sidecar for the driver

Generally they are run as a part of the same pod